### PR TITLE
Add swmm-toolkit

### PR DESCRIPTION
**Pull request by cheginit**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by cheginit** [comment]
@karosc Please confirm that you agree to be a maintainer of this package.

**Comment by cheginit** [comment]
@karosc All builds are passing except for `osx`, can you please look at the logs [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1041948&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7&t=fa7b4b17-b6ff-5c9c-8cfc-15f888c92310&l=1964) and see if you can figure out the issue?

**Comment by karosc** [comment]
@cheginit, this build process can be finicky with versions. IMO, updating the compiler and OMP versions shouldn't break things ,but sometimes ti does. I ran the build process on github tonight to regenerate the logs and examine the github environment. At the very least:

1. github is using [AppleClang 14](https://github.com/karosc/swmm-python/actions/runs/11117145656/job/30888602922#step:3:256) while Azure is using [Clang 17](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1041958&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7&t=fa7b4b17-b6ff-5c9c-8cfc-15f888c92310&l=1542)
2.  github is using [OMP 5.0](https://github.com/karosc/swmm-python/actions/runs/11117145656/job/30888602922#step:3:446) while azure is using [OMP 5.1](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1041958&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7&t=fa7b4b17-b6ff-5c9c-8cfc-15f888c92310&l=1700)

Maybe you can try downgrading those packages and see if that helps?

**Comment by karosc** [comment]
Also, should the openmp libs be in [the hosts section](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#host)? I'm not sure.

**Comment by cheginit** [comment]
> Maybe you can try downgrading those packages and see if that helps?

@karosc I don't think version is the issue, since I can build it on my machine with clang 16 and openmp 5.1, without any issue.

> Also, should the openmp libs be in [the hosts section](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#host)? I'm not sure.

I am not sure. Some example meta files have it in their host and some don't. I think I tried it in one of the earlier versions. I will give it a try now since only osx is not working.

**Comment by cheginit** [comment]
@karosc So, I finally figured out the correct way of setting compiler version for osx and set it to 14. It didn't solve the issue and got the same error as before. Any other ideas?

**Comment by karosc** [comment]
@cheginit, I did some testing on my own fork after merging in yours. I got mac to build by:

1. reverting to the newer build tools (removing the conda_build_config.yml)
2. Adding _FORTIFY_SOURCE=2 to the CFLAGS in the meta.yml. 
3. Removing llvm-openmp from osx build deps. Theoretically the build process should build omp from source on macosx. I  was worried the conda version was interfering with the process, but not sure if this made a difference. It works without it anyway.

See my [updated meta.yml for reference](https://github.com/karosc/staged-recipes/blob/swmm-toolkit/recipes/swmm-toolkit/meta.yaml#L20-L21)

It seems conda-build is setting the _FORTIFY_SOURCE option in way that interferes with our build process. The correct way to deal with this would probably be to undef and redef it in the cmake lists in the source code.....but that require a release of a new version, which we are not prepared to do right now due to issues with the github build environment interfering with our regression tests.

Setting the CFLAGS in the meta.yml does get the mac version to build, but there are tons of redefine macro warnings in the [build logs](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1043667&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7). We could add `-Wno-macro-redefined` to our CFLAGS to suppress the warnings, but it maybe should leave them to remind us this is kind of a hack...?

Thanks for re-invigorating this effort! If you can get this to lint and build in this PR, we can discuss adding docs in the main repo. There is something to be said about building SWMM from source and model output reproducibility.

**Comment by cheginit** [comment]
@karosc Awesome! I added the flags and it did fix the issue. What you said make sense to me. Thanks for committing to make an effort in fixing the issue. I kept the `llvm-openmp` just to make the build a bit faster and since it's a common approach in other meta files that support `osx`.

**Comment by cheginit** [comment]
@conda-forge/help-python-c, this is ready for review, thanks! Also, can adding `osx-arm64` support be done through the automated way of editing the [`osx_arm64.txt`](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/migrations/osx_arm64.txt) file after this PR gets merged?

File: recipes/swmm-toolkit/meta.yaml
Comment (xhochy) [inline recipes/swmm-toolkit/meta.yaml:21]: This overrides and not extends rhe flags, it should extend.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/swmm-toolkit/meta.yaml
Comment (cheginit) [reply recipes/swmm-toolkit/meta.yaml:21]: Right, for now that's what we want, until we take care of the FORTIFY_SOURCE issue in the next release. Do you mean something like this in `script_env`: `CFLAGS="${CFLAGS} -D_FORTIFY_SOURCE=2"`

Comment (xhochy) [reply recipes/swmm-toolkit/meta.yaml:21]: Yes, exactly that.

Comment (cheginit) [reply recipes/swmm-toolkit/meta.yaml]: Done.

Comment (cheginit) [reply recipes/swmm-toolkit/meta.yaml]: The `osx` build throws this error: `export: '-D_FORTIFY_SOURCE=2': not a valid identifier`. Any ideas how to solve this? Maybe this cannot be done with `script_env`?

Comment (cheginit) [reply recipes/swmm-toolkit/meta.yaml]: @xhochy I fixed it by moving them to `script`.

File: recipes/swmm-toolkit/meta.yaml
Comment (xhochy) [inline recipes/swmm-toolkit/meta.yaml:36]: These two should also be in `host`
```suggestion
    - llvm-openmp  # [osx]
    - libgomp  # [linux]
```

**Review by xhochy — APPROVED** [review]

**Pull request merged**
